libs/common/types.hpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_COMMON_HPP
+#define IROHA_COMMON_HPP
+
+#include <dirent.h>
+#include <array>
+#include <crypto/base64.hpp>
+#include <cstdio>
+#include <gsl/gsl_byte>
+
+
+/**
+ * This file defines common types used in iroha.
+ *
+ * std::string is convenient to use but it is not safe, because we can not
+ * guarantee at compile-time fixed length of the string.
+ *
+ * For std::array it is possible, so we prefer it over std::string.
+ */
+namespace iroha {
+
+  using byte_t = uint8_t;
+
+  /**
+   * Base type which represents blob of fixed size.
+   */
+  template <size_t size_>
+  class blob_t : public std::array<byte_t, size_> {
```

Comment:
Copying is ok at this case, because `blob_t` neither add new members, nor use pointers.
`blob_t` is just `std::array` but with extended interface, so it is safe.